File: numerical_method/languageComparison/primes/prime.jl
upper_bound = 100000
counter = 0
primes = []

for i in 2:upper_bound
	is_prime = true
	for j in 2:i-1
		if i % j == 0
			is_prime = false
			break
		end
	end
	if is_prime
		push!(primes, i)
		global counter += 1
	end
end 

println("$counter primes found under $upper_bound")

File: numerical_method/languageComparison/primes/prime.f90
program prime
use OMP_LIB

    implicit none
    integer :: i, j, n, count
    logical :: isprime
    integer, allocatable :: primes(:)
    integer :: upper_bound = 100000
    allocate(primes(upper_bound/2))
    count = 0
    !$omp parallel do private(isprime, i) shared(primes) reduction(+:count)
    do n = 2, upper_bound ! this is inclusive
        isprime = .true.
        do i = 2, n-1
            if (mod(n, i) == 0) then
                isprime = .false.
                exit
            end if
        end do
        if (isprime) then
            count = count + 1
            primes(count) = n
        end if
    end do
    !$omp end parallel do

    print *, "Total number of primes found:", count

end program prime
